### DISCUSSION TITLE
Recovering costs to run the IOCCC

### DISCUSSION THREAD

**Discussion by lcn2** [ioccc-src/temp-test-ioccc#2278] [Ideas]
## IOCCC costs

There is a run rate for IOCCC web servers, GitHub Pro, and hiring folk.  There are some $-ive costs for running the IOCCC.

For all the years, the IOCCC judges have paid for the infrastructure (servers, server hosting, domain registration, laptops, development tools, service accounts, etc.) out of our own pockets.   From time to time we hire a contractor to help for things like web site design.  The IOCCC has grown to the extent that it would be nice to generate a little revenue to recover some of those costs.  

We are **NOT** looking for turning the IOCCC into a business.  We simply want to get something to cover the on-going costs associated with the IOCCC. 

Nearly all of the IOCCC costs paid for by one IOCCC judge.  Now that IOCCC judge is retired and on fixed income.  There is an effort to keep costs down, but nevertheless there are costs.

## Option Patreon

We could recover costs via something like Patreon. That has its own issues including the need to keep a Patreon supporter happy and informed.  We would need to describe what people get with Patron tiers, keep those subscribing up to date and happy, as well as to ask for people to become a Patreon.  Care for a Patreon group is extra overhead that we might **NOT** want to take on.

## Option Google ads

We could recover costs via something like Google ads. Adding google ads would be easy to do programmatically. And perhaps there is a way for add them without making them too intrusive. We know some people do not like ads and use ad blockers. That would be OK if some people did that .. we won't try to force Google ad on people who block them.

## Option something else

Or we could do something else.

## Option do nothing

... well that is always an option :-) but that does not solve the problem :-(

# What do you think?

Comments, questions and concerns are welcome.

**Comment by xexyl** [comment]
I think ads is a problem in general and to encourage google ads is even bigger as they have been known to spread malvertisements. Also they allow websites to look legitimate but they aren't. I saw a website in google ads that looked authentic (if not careful) for GIMP but it was malicious. They allow this is my understanding.

So I would strongly recommend against ads. And yeah people use ad blockers and some of us use DNS blacklisting too (that implies I do that). But still they are both intrusive and dangerous esp with google (and probably others too).

I'm not sure how Patreon works but it seems wholly better than a service that wilfully allows people to manipulate how they look to spread malware. Even if we wouldn't do that (of course) it's not good to encourage them or use such a service I think.

I have some screenshots of the above but of course it's likely easy to find cases of their ads spreading malware. I wouldn't trust google at all for this whatever the costs might be unless it's absolutely impossible not to.

A question though that might help. What ARE the costs? Depending on the costs maybe some of us can donate a bit at times at least if not more regularly?

**Comment by SirWumpus** [comment]
> Option Google ads

Nuke with extreme prejudice.

> Option Patreon

I support a content creator, The Buur Pit, that make weekly videos for `Elite Dangerous`.  The low tier is 5$/month and they have something like 150 supporters just now.  As what to offer:

* Stats on contest entries that have remained a mystery for years (other than paper shredded).
* More judge bio/background.
* Where are people seeing obfuscation in real life (outside of politics).
* Video (blooper reel) of judging in progress plus commentary.
* Are there stories / places where winners appear for literature or teaching or launching pad for real world projects?  (eg. Don Libes' book `Obfuscated C and Other Mysteries`).
* Interviews with past winners.
* If Monty Python were a programmer... (comic strip or humour article like [BOFH](https://www.theregister.com/offbeat/bofh/).
* Monty Python vs Mr Bean code fest?  (not sure what i had in mind with this pseudo title).

Monthly updates might be difficult, maybe Quarterly?

> Option something else

* For a long time when I first offered my Sendmail `milters` for free, I encouraged people to say "thanks" by purchasing things from an Amazon Wishlist.  While not a direct form of payment, it does reward the Judges so they can continue doing what they enjoy.
* Annual donation drives?  Similar to NetBSD.org or Wikipedia donation drives.
* Sponsorship from other .orgs? Usenix, IEEE/ACM, [Yuk Yuks](https://yukyuks.com/), maybe a university math/computing dept., Indian restaurants.
* Write and sell a book about the IOCCC.  Its been running a long time.  There must be a wealth of history to tell from inception (which you already overview on the web site), to judging, IOCCC presentations, impact on life and views about tech. (and the World beyond Hello), etc.

**Reply by SirWumpus** [reply]
Add to my suggestions list of ways to thank the judges:

* Sometimes people do not want to give money for assorted reasons, like how a envelope of money for a birthday or Xmas is so impersonal.  So on top of Amazon wish list, maybe a list of the Judges' preferred restaurants (offering gift cards (I know)) for which they can be "sent" for a meal out.  Or maybe intl. pizza / curry / chinese / etc. food delivery :-D  Something more targetted than an arbitrary donation.

**Reply by xexyl** [reply]
> Add to my suggestions list of ways to thank the judges:
> 
> * Sometimes people do not want to give money for assorted reasons, like how a envelope of money for a birthday or Xmas is so impersonal.  So on top of Amazon wish list, maybe a list of the Judges' preferred restaurants (offering gift cards (I know)) for which they can be "sent" for a meal out.  Or maybe intl. pizza / curry / chinese / etc. food delivery :-D  Something more targetted than an arbitrary donation.

European chocolate!

**Comment by xexyl** [comment]
> > Option Google ads
> 
> Nuke with extreme prejudice.

I'm glad I'm not the only one who agrees with this!

> 
> > Option Patreon
> 
> I support a content creator, The Buur Pit, that make weekly videos for `Elite Dangerous`. The low tier is 5$/month and they have something like 150 supporters just now. As what to offer:
> 
> * Stats on contest entries that have remained a mystery for years (other than paper shredded).
> * More judge bio/background.
> * Where are people seeing obfuscation in real life (outside of politics).
> * Video (blooper reel) of judging in progress plus commentary.
> * Are there stories / places where winners appear for literature or teaching or launching pad for real world projects?  (eg. Don Libes' book `Obfuscated C and Other Mysteries`).
> * Interviews with past winners.
> * If Monty Python were a programmer... (comic strip or humour article like [BOFH](https://www.theregister.com/offbeat/bofh/).
> * Monty Python vs Mr Bean code fest?  (not sure what i had in mind with this pseudo title).
> 
> Monthly updates might be difficult, maybe Quarterly?
> 
Some are these are fun though it might be nice for people who can't afford a monthly or quarterly payment too. But even so it could be an incentive to people for sure. Nice idea.

> > Option something else
> 
> * For a long time when I first offered my Sendmail `milters` for free, I encouraged people to say "thanks" by purchasing things from an Amazon Wishlist.  While not a direct form of payment, it does reward the Judges so they can continue doing what they enjoy.

That's a great idea. I've actually pondered getting some DVDs for Landon - a brilliant sitcom (The IT Crowd). But I've not mentioned it for a variety of reasons. But might as well now.


> * Write and sell a book about the IOCCC.  It's been running a long time.  There must be a wealth of history to tell from inception (which you already overview on the web site), to judging, IOCCC presentations, impact on life and views about tech. (and the World beyond Hello), etc.

I really like this. I once told Landon he should do something like that with his life story too. I'd definitely buy it and I'd definitely buy a book by the IOCCC judges about the IOCCC - or probably anything else for that matter. `*hint*, *hint*`

**Reply by SirWumpus** [reply]
> That's a great idea. I've actually pondered getting some DVDs for Landon - a brilliant sitcom (The IT Crowd). But I've not mentioned it for a variety of reasons. But might as well now.

Maybe all 4 seasons of `Halt And Catch Fire` on DVD/BluRay?  (Hmm. Landon strikes me a [Laser Disc](https://en.wikipedia.org/wiki/LaserDisc) guy ;-) )

**Comment by xexyl** [comment]
> > That's a great idea. I've actually pondered getting some DVDs for Landon - a brilliant sitcom (The IT Crowd). But I've not mentioned it for a variety of reasons. But might as well now.
> 
> Maybe all 4 seasons of `Halt And Catch Fire` on DVD/BluRay? (Hmm. Landon strikes me a [Laser Disc](https://en.wikipedia.org/wiki/LaserDisc) guy ;-) )

Afraid I don't know the show - I just know the phrase and the CPU instruction (though I couldn't name the CPU or CPUs that has/have it). I don't know if The IT Crowd is available in that format though. Is Halt and Catch Fire a good show? Well I'm sure it is in your view as otherwise you probably wouldn't be suggesting it :-) .. well the reason I mentioned IT Crowd is it's brilliant and I referred to it once and he didn't know what it was. Also I discovered a wonderful Easter egg on the first series (or if you prefer season) DVD esp for the first episode.

**Reply by SirWumpus** [reply]
[Halt And Catch Fire](https://en.wikipedia.org/wiki/Halt_and_Catch_Fire_(TV_series)) is a historical fiction (much like a Guy Gavriel Kay's books) that roughly parallels the growth of the micro computer industry through to the Web.  I found it to a very good series, but then I grew up in that era of tech. and enjoy that sort of thing.

**Comment by xexyl** [comment]
> [Halt And Catch Fire](https://en.wikipedia.org/wiki/Halt_and_Catch_Fire_(TV_series)) is a historical fiction (much like a Guy Gavriel Kay's books) that roughly parallels the growth of the micro computer industry through to the Web. I found it to a very good series, but then I grew up in that era of tech. and enjoy that sort of thing.

As far as being set in the 80s (as I think I saw)...

https://github.com/ioccc-src/temp-test-ioccc/assets/53008573/abc3fdda-085a-495c-bf25-47253770507d

Make sure to have sound on of course. I was going to do a few screenshots with subtitles but it'd be a tall image, maybe too tall. It's from one of my favourite episodes - The Work Outing.

Not that I have a thing against the 80s. It's when I first got a computer and my favourite band was formed in the 80s too. It just reminded me of this scene since I had brought up the show.

**Reply by SirWumpus** [reply]
I'm always amazed when something in life coincides with current discussions, someone on /. this morning...
[How CP/M Launched the Next 50 Years of Operating Systems](https://computerhistory.org/blog/fifty-years-of-the-personal-computer-operating-system/).  This is a perfect example of book worthy material; surely IOCCC has similar stories.

As for the 1980's I enjoyed much of the New Wave, Synth-Pop, Electronic music, like `Ultravox`, `The Human League`, `Devo`, `Tangerine Dream` (would have to troll my CD collection for more).  Certainly beats opera or rap music IMO.

I think I recall seeing or hearing about `The It Crowd` it from someone else (probably some of my British tech. friends).  Never got into it back then, but might try to find it on Roku or BritBox streaming.

Has anyone considered nominating the IOCCC for the Computer History Museum?

**Reply by xexyl** [reply]
> I'm always amazed when something in life coincides with current discussions, someone on /. this morning... [How CP/M Launched the Next 50 Years of Operating Systems](https://computerhistory.org/blog/fifty-years-of-the-personal-computer-operating-system/). This is a perfect example of book worthy material; surely IOCCC has similar stories.
> 

Tolkien used the word 'applicability' for that as it indeed happens a lot.

> As for the 1980's I enjoyed much of the New Wave, Synth-Pop, Electronic music, like `Ultravox`, `The Human League`, `Devo`, `Tangerine Dream` (would have to troll my CD collection for more). Certainly beats opera or rap music IMO.
> 
Metal for me. Lifesaving for me. My mum saw the Beatles more than once. So did my dad. By New Wave do you mean for instance New Wave of British Heavy Metal or something else? That's where I am.

> I think I recall seeing or hearing about `The It Crowd` it from someone else (probably some of my British tech. friends). Never got into it back then, but might try to find it on Roku or BritBox streaming.
> 
It's brilliant and you would understand all the jokes. Even those who don't though can really get a lot of laughs from it. The episode I gave a scene from is an example but that episode is constant laughter. Not that many tech jokes there but many do have them. I won't say more in case you decide to watch except I might suggest you find information on 'The Speech' if you want a nice introduction (though it's not the first year) and in particular the box that they trick their boss into believing is the Internet. But of course there are plenty of other good episodes that can be an introduction including the first. The Easter egg on the DVD is that the second English track has for the first episode has not what is being said but what is being said in computer speak. Sometimes in unix commands, sometimes in assembly (for Windows) and other times elite speak (which in this can be funny). Probably other things too. The next five episodes it's not so clever sadly.


> Has anyone considered nominating the IOCCC for the Computer History Museum?

I like that idea though I'm not sure how it would be presented. Maybe Landon has some ideas? I know he's been there before, seeing the [Original Tinkertoy Computer](https://www.computerhistory.org/collections/catalog/X39.81).

Appreciate the dialogue and tips and interesting thoughts!

**Reply by SirWumpus** [reply]
> It's brilliant and you would understand all the jokes. 

Well given I had the fortune to do my senior school in Australia in the 1980's, British style of humour was very (probably still is) popular combined with comput-er-ing, programming, and many late nights using `ddt` to figure out Z80 assembler and stuff I'm pretty sure the humour would not be lost on me.

**Reply by xexyl** [reply]
> > It's brilliant and you would understand all the jokes.
> 
> Well given I had the fortune to do my senior school in Australia in the 1980's, British style of humour was very (probably still is) popular combined with comput-er-ing, programming, and many late nights using `ddt` to figure out Z80 assembler and stuff I'm pretty sure the humour would not be lost on me.

It definitely would not be! I have shown episodes to people who aren't computer people and they got a lot of laughs too.

**Reply by SirWumpus** [reply]
One can get a lot of good laughs from looking at hair and clothing styles from movies and music videos from early 1980's.

**Reply by xexyl** [reply]
> One can get a lot of good laughs from looking at hair and clothing styles from movies and music videos from early 1980's.

Do you have any examples?

There are numerous films that I enjoy from the 80s but I have never paid attention to style. Not in films and not otherwise. Unless it's REALLY FAR out there I guess. But that's just a fleeting thing. 

I do love Die Hard (though the first is not my favourite) and Lethal Weapon (though not my favourite of them - though the jump scene and the sniper scene are amongst the best in the series for me). 


My favourite film is more recent. Iirc 2017. The Hitman's Bodyguard with Samuel L Jackson and Ryan Reynolds. I quite enjoy the sequel as well. 

Anything with Jackson is great. My favourite Die Hard film is the third and I really find his character funny (it's hard to say whether Bruce Willis's character is funnier but the dynamic between the two is even better kind of like with Jackson and Reynolds and their characters). 

But that's all besides the point I suppose. If you have an example though I am certainly curious.

**Reply by SirWumpus** [reply]
Its been a while.  I suppose `Pretty In Pink`, `Can't Buy Me Love`, `Electric Dreams`, `Footloose` (1984).  Maybe some others that come before 1985.

**Reply by xexyl** [reply]
Ah. Well I don't know those films (to be fair I don't know a lot of films that most people do as I tend to prefer reading) so maybe that's also part of it. Thanks.

**Reply by SirWumpus** [reply]
The blown back hair, hair spray galore, pastels, jacket shoulder pads (men & women).

**Reply by xexyl** [reply]
The closest I come to knowing any of that is Hairspray but only the play. And only because my mum has seen it. But that's not the same thing. 

Thanks. I probably will never see them but maybe if I come across them I'll consider it.

**Comment by lcn2** [comment]
Your feedback @SirWumpus and @xexyl , along with other feedback has been noted.  As do not plan to run Google ads for the moment.

**Reply by xexyl** [reply]
> Your feedback @SirWumpus and @xexyl , along with other feedback has been noted.  As do not plan to run Google ads at this point.

Thank you! That's great news.
